code/test/tokens.js

```diff
@@ -0,0 +1,472 @@
+    // await tokenTransfer(
+    //   {
+    //     wallet: alice,
+    //     operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+    //     tokenType: ONEConstants.TokenType.ERC1155,
+    //     contractAddress: testerc1155.address,
+    //     tokenId: 8,
+    //     dest: bob.wallet.address,
+    //     amount: 3
+    //   }
+    // )
+  })
+})
+
+// makeWallet uses an index and unlocked web3.eth.account and creates and funds a ONEwallet
+const makeWallet = async (accountIndex, lastResortAddress) => {
+  // create wallet
+  const EFFECTIVE_TIME = Math.floor(Date.now() / INTERVAL / 6) * INTERVAL * 6 - DURATION / 2
+  const TEN_ETH = unit.toWei('10', 'ether')
+  const { wallet, seed, hseed, root, client: { layers } } = await TestUtil.createWallet({
+    salt: new BN(accountIndex),
+    effectiveTime: EFFECTIVE_TIME,
+    duration: DURATION,
+    maxOperationsPerInterval: SLOT_SIZE,
+    lastResortAddress,
+    spendingLimit: TEN_ETH
+  })
+  // Fund wallet
+  await web3.eth.sendTransaction({
+    from: lastResortAddress,
+    to: wallet.address,
+    value: TEN_ETH
+  })
+  const state = await CheckUtil.getONEWalletState(wallet)
+  return { wallet, seed, hseed, root, layers, lastResortAddress, state }
+}
+
+// makeTokens makes test ERC20, ERC20Decimals9, ERC721, ERC1155
+const makeTokens = async (owner) => {
+  // create an ERC20
+  const TESTERC20 = artifacts.require('TestERC20')
+  const testerc20 = await TESTERC20.new(10000000, { from: owner })
+  // create an ERC20Decimals9
+  const TESTERC20DECIMALS9 = artifacts.require('TestERC20Decimals9')
+  const testerc20d9 = await TESTERC20DECIMALS9.new(10000000, { from: owner })
+  // create an ERC721
+  const TESTERC721 = artifacts.require('TestERC721')
+  const tids = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  const uris = ['ipfs://test721/0', 'ipfs://test721/1', 'ipfs://test721/2', 'ipfs://test721/3', 'ipfs://test721/4', 'ipfs://test721/5', 'ipfs://test721/6', 'ipfs://test721/7', 'ipfs://test721/8', 'ipfs://test721/9']
+  const testerc721 = await TESTERC721.new(tids, uris, { from: owner })
+  // create an ERC1155
+  const TESTERC1155 = artifacts.require('TestERC1155')
+  const tids1155 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  const amounts1155 = [10, 20, 20, 20, 20, 20, 20, 20, 20, 100]
+  const uris1155 = ['ipfs://test1155/0', 'ipfs://test1155/1', 'ipfs://test1155/2', 'ipfs://test1155/3', 'ipfs://test1155/4', 'ipfs://test1155/5', 'ipfs://test1155/6', 'ipfs://test1155/7', 'ipfs://test1155/8', 'ipfs://test1155/9']
+  const testerc1155 = await TESTERC1155.new(tids1155, amounts1155, uris1155, { from: owner })
+  // create a TokenTracker
+  const TOKENTRACKER = artifacts.require('TokenTracker')
+  const tokentracker = await TOKENTRACKER.new()
+  return { testerc20, testerc20d9, testerc721, testerc1155, tokentracker }
+}
+
+// tokenTransfer commits and reveals a wallet transaction
+const tokenTransfer = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount }) => {
```

Comment:
Let's call it assetTransfer, to cover both native asset (ONE) from tokens, and in general distinguish the two

Reply:
done